### PR TITLE
Small fixes to generate the dataset labels and train the models

### PR DESCRIPTION
**Pull request by anthonyhu**
This PR fixes a few issue when generating the dataset labels and training the models.

In `render_polygon`, sometimes the list of polygon is empty, in that case nothing should be done right?

### CODE REVIEW

**Comment by tom-roddick** [comment]
Hi @anthonyhu thanks for the fixes, looks great! The only change to the PR I'd like to make is that the NuScenes train/val splits should be kept the same as the ones used in the paper. Unfortunately the official splits are invalid for the map prediction task since the ego-vehicle visits the same stretch of road in both train and val, allowing the network to overfit. Other than that I'm happy to accept the PR :)

**Comment by anthonyhu** [comment]
Hey Tom! I see - I had changed the val split in a former commit but in this PR I'm not touching it (`src/data/nuscenes/splits.py` is unchanged) :)

**Comment by tom-roddick** [comment]
Ah my apologies - didn't see the commit which changed them back!  Accepting the PR now :)
